server/src/main/java/org/opensearch/common/io/stream/NamedWriteableRegistryParseRequest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.io.stream;
+
+import org.opensearch.transport.TransportRequest;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * Extensibility support for Named Writeable Registry: request to extensions to parse context
+ *
+ * @opensearch.internal
+ */
+public class NamedWriteableRegistryParseRequest extends TransportRequest {
+
+    private final Class categoryClass;
+    private byte[] context;
+
+    /**
+     * @param categoryClass Class category for this parse request
+     * @param context StreamInput object to convert into a byte array and transport to the extension
+     * @throws IllegalArgumentException if context bytes could not be read
+     */
+    public NamedWriteableRegistryParseRequest(Class categoryClass, StreamInput context) {
+        try {
+            byte[] streamInputBytes = context.readAllBytes();
+            this.categoryClass = categoryClass;
+            this.context = Arrays.copyOf(streamInputBytes, streamInputBytes.length);
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Invalid context", e);
+        }
+    }
+
+    /**
+     * @param in StreamInput from which class fields are read from
+     * @throws IllegalArgumentException if the fully qualified class name is invalid and the class object cannot be generated at runtime
+     */
+    public NamedWriteableRegistryParseRequest(StreamInput in) throws IOException {
+        super(in);
+        try {
```

Comment:
Same as above

Reply:
Yes because we send the fully qualified class names between OpenSearch and the SDK. Within this constructor, we try to convert the fully qualified class name into a class instance to register. If one node does not have the class definition, we cannot register the entry with this category class, so we catch the ClassNotFound exception and throw an IllegalArguementException along with the original exception message